src/HDInsight/HDInsight/Models/Management/AzureHdInsightManagementClient.cs

```diff
@@ -225,5 +226,21 @@ public virtual void RestartHosts(string resourceGroupName, string clusterName, I
         {
             HdInsightManagementClient.VirtualMachines.RestartHosts(resourceGroupName, clusterName, hosts);
         }
+        
+        private void ResetClusterIdentity(ClusterCreateParametersExtended createParams, string cloudAadAuthority, string cloudDataLakeAudience)
+        {
+            var configuation = (Dictionary<string, Dictionary<string, string>>)createParams.Properties.ClusterDefinition.Configurations;
+            Dictionary<string, string> clusterIdentity;
+            if(!configuation.TryGetValue("clusterIdentity", out clusterIdentity))
+            {
+                return;
+            }
+            clusterIdentity.SetProperty("clusterIdentity.resourceUri", cloudDataLakeAudience);
```

Comment:
Nit - you can just say `clusterIdentity["clusterIdentity.resourceUri"] = cloudDataLakeAudience;` for simplicity. BTW, what if `clusterIdentity["clusterIdentity.resourceUri"]` is not set originally?

Reply:
> Nit - you can just say `clusterIdentity["clusterIdentity.resourceUri"] = cloudDataLakeAudience;` for simplicity. BTW, what if `clusterIdentity["clusterIdentity.resourceUri"]` is not set originally?

Fixed. If it is not set originally, then our code will add the key/value pair, I think it is ok.